src/Sarif/Writers/SarifLogger.cs

```diff
@@ -356,7 +367,16 @@ private void CaptureFile(Uri uri)
                 return;
             }
 
-            _run.Files[fileDataKey] = FileData.Create(uri, _loggingOptions);
+
+            Encoding encoding = null;
+
+            try
+            {
+                encoding = Encoding.GetEncoding(_run.DefaultFileEncoding);
+            }
+            catch (ArgumentException) { }
```

Comment:
@EasyRhin0 Also, what does `Encoding.GetEncoding` do it the argument is `null`? If if throws `ArgumentNullException`, that's fine, because `ArgumentException` will catch it.

Reply:
It does throw ArgumentNullException for any

Reply:
Test added